datafusion/core/tests/sql/group_by.rs

```diff
@@ -905,3 +905,220 @@ async fn csv_query_group_by_order_by_avg_group_by_substr() -> Result<()> {
     assert_batches_sorted_eq!(expected, &actual);
     Ok(())
 }
+
+#[tokio::test]
+async fn query_group_by_with_filter() -> Result<()> {
+    let schema = Arc::new(Schema::new(vec![
+        Field::new("c1", DataType::Int32, true),
+        Field::new("c2", DataType::Int32, true),
+    ]));
+
+    let data = RecordBatch::try_new(
+        schema.clone(),
+        vec![
+            Arc::new(Int32Array::from(vec![
+                Some(1),
+                Some(1),
+                Some(2),
+                Some(2),
+                Some(3),
+            ])),
+            Arc::new(Int32Array::from(vec![
+                Some(10),
+                Some(20),
+                Some(10),
+                Some(20),
+                Some(10),
+            ])),
+        ],
+    )?;
+
+    let ctx = use_postgres_dialect();
+    ctx.register_batch("test", data)?;
+    let sql =
```

Comment:
I think another case that is important is a query with an aggregate that both does / does not have a filter

For example


```sql
SELECT c1, 
  SUM(c2) FILTER (WHERE c2 >= 20) as result,
  SUM(c2)  as result_no_filter,
FROM test GROUP BY c1